sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamFileCatalog.scala

```diff
@@ -54,6 +54,6 @@ class StreamFileCatalog(sqlContext: SQLContext, path: Path) extends FileCatalog
   override def refresh(): Unit = {}
 
   override def allFiles(): Seq[FileStatus] = {
-    fs.listStatus(metadataLog.get(None, None).flatMap(_._2).map(new Path(_)))
+    fs.listStatus(metadataLog.allFiles().map(f => new Path(f.path)))
```

Comment:
We only need to store the ones that we need later on:

Need:
 - long length
 - Path path

Might need:
 - long modification_time

Obvious:
 - boolean isdir

Not sure (but not used now):
 - int block_replication
 - long blocksize

Don't need
 - FsPermission permission
 - String owner
 - String group
 - long access_time